hpedockerplugin/volume_manager.py

```diff
@@ -1485,48 +1491,79 @@ def unmount_volume(self, volname, vol_mount, mount_id):
             # by some other node, it can go to that different ETCD root to
             # fetch the volume meta-data and do the cleanup.
             if self._node_id not in node_mount_info:
-                return json.dumps({u"Err": "Volume '%s' is mounted on another"
-                                           " node. Cannot unmount it!" %
-                                           volname})
-
-            LOG.info("node_id '%s' is present in vol mount info"
-                     % self._node_id)
-
-            mount_id_list = node_mount_info[self._node_id]
-
-            LOG.info("Current mount_id_list %s " % mount_id_list)
-
-            try:
-                mount_id_list.remove(mount_id)
-            except ValueError as ex:
-                pass
+                if 'old_path_info' in vol:
+                    LOG.info("Old path info present in volume: %s"
+                             % path_info)
+                    for pi in vol['old_path_info']:
+                        node_id = pi[0]
+                        if node_id == self._node_id:
```

Comment:
Loop is required because it is possible to migrate volume from Node1 to Node2 and then to Node3 in which case there are going to be two entries under "old_path_info" - one for Node1 and other for Node2 that would need iteration.

Reply:
Thanks @imran-ansari , it answers my other question, that if a node moves from N1->N2, N2->N3, and when unmount happens on N1,N2 also.